spring-rabbit/src/test/java/org/springframework/amqp/rabbit/log4j2/ExtendAmqpAppenderTests.java

```diff
@@ -0,0 +1,252 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.log4j2;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.amqp.core.BindingBuilder;
+import org.springframework.amqp.core.FanoutExchange;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageDeliveryMode;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.connection.RabbitConnectionFactoryBean;
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.rabbit.junit.RabbitAvailable;
+import org.springframework.amqp.rabbit.junit.RabbitAvailableCondition;
+import org.springframework.amqp.utils.test.TestUtils;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.test.util.ReflectionTestUtils;
+
+/**
+ * @author Francesco Scipioni
+ *
+ * @since 2.2.4
+ */
+@RabbitAvailable
+public class ExtendAmqpAppenderTests {
+
+	private static final LoggerContext LOGGER_CONTEXT = (LoggerContext) LogManager.getContext(false);
+
+	private static final URI ORIGINAL_LOGGER_CONFIG = LOGGER_CONTEXT.getConfigLocation();
+
+	@BeforeAll
+	public static void setup() throws IOException {
+		LOGGER_CONTEXT.setConfigLocation(new ClassPathResource("log4j2-extend-amqp-appender.xml").getURI());
+		LOGGER_CONTEXT.reconfigure();
+	}
+
+	@AfterAll
+	public static void teardown() {
+		LOGGER_CONTEXT.setConfigLocation(ORIGINAL_LOGGER_CONFIG);
+		LOGGER_CONTEXT.reconfigure();
+		RabbitAvailableCondition.getBrokerRunning().deleteQueues("log4jTest", "log4j2Test");
+		RabbitAvailableCondition.getBrokerRunning().deleteExchanges("log4j2Test", "log4j2Test_uri");
+	}
+
+	@Test
+	public void test() {
+		CachingConnectionFactory ccf = new CachingConnectionFactory("localhost");
```

Comment:
Would you mind to `destroy()` this factory in the end of test to avoid resources leaking?

Reply:
Should I also fix this in `spring-rabbit/src/test/java/org/springframework/amqp/rabbit/log4j2/AmqpAppenderTests.java` ?

Reply:
Yes, please. I believe not closed `ConnectionFactory` leads to resource leaking.

Thanks